pkg/oc/generate/app/cmd/resolve.go

```diff
@@ -184,14 +187,25 @@ func Resolve(appConfig *AppConfig) (*ResolvedComponents, error) {
 
 // AddSourceRepositoriesToRefBuilder adds the provided repositories to the reference builder, identifies which
 // should be built using Docker, and then returns the full list of source repositories.
-func AddSourceRepositoriesToRefBuilder(b *app.ReferenceBuilder, repos []string, g *GenerationInputs) (app.SourceRepositories, error) {
+func AddSourceRepositoriesToRefBuilder(b *app.ReferenceBuilder, c *ComponentInputs, g *GenerationInputs, s, i *[]string) (app.SourceRepositories, error) {
 	strategy := g.Strategy
 	if strategy == generate.StrategyUnspecified {
 		strategy = generate.StrategySource
 	}
-	for _, s := range repos {
-		if repo, ok := b.AddSourceRepository(s, strategy); ok {
-			repo.SetContextDir(g.ContextDir)
+	// when git is installed we keep default logic. sourcelookup.go will do sorting of images, repos
+	if git.IsGitInstalled() || len(c.SourceRepositories) > 0 {
```

Comment:
if git is installed all logic in componentInputs and flag works. If it does not exist logic can recognize only URL and sets it as SourceRepository. If we pass hostDir - we need logic below.